src/commands/math/basicSymbols.ts

```diff
@@ -266,33 +303,54 @@ optionProcessors.autoParenthesizedFunctions = function (cmds) {
   return dict;
 }
 
-var Letter = P(Variable, function(_, super_) {
-  _.init = function(ch) { return super_.init.call(this, this.letter = ch); };
-  _.checkAutoCmds = function (cursor) {
+class Letter extends Variable {
+  letter:string;
+  
+  constructor (ch:string) {
+    super(ch);
+    this.letter = ch;
+  };
+  checkAutoCmds (cursor:Cursor) {
     //handle autoCommands
-    var autoCmds = cursor.options.autoCommands, maxLength = autoCmds._maxLength;
+    var autoCmds = cursor.options.autoCommands;
+    var  maxLength = autoCmds._maxLength || 0;
     if (maxLength > 0) {
       // want longest possible autocommand, so join together longest
       // sequence of letters
-      var str = '', l = this, i = 0;
+      var str = '';
+      var l:NodeRef = this;
+      var i = 0;
       // FIXME: l.ctrlSeq === l.letter checks if first or last in an operator name
       while (l instanceof Letter && l.ctrlSeq === l.letter && i < maxLength) {
-        str = l.letter + str, l = l[L], i += 1;
+        str = l.letter + str;
+        l = l[L];
+        i += 1;
       }
       // check for an autocommand, going thru substrings longest to shortest
       while (str.length) {
         if (autoCmds.hasOwnProperty(str)) {
-          for (var i = 1, l = this; i < str.length; i += 1, l = l[L]);
-          Fragment(l, this).remove();
-          cursor[L] = l[L];
-          return LatexCmds[str](str).createLeftOf(cursor);
+          l = this;
+          for (i = 1; l && i < str.length; i += 1, l = l[L]);
+
+          new Fragment(l as MQNode, this).remove();
+          cursor[L] = (l as MQNode)[L];
+
+          var cmd = LatexCmds[str];
+          var node;
+          if (isMQNodeClass(cmd)) {
+            node = new (cmd as typeof TempSingleCharNode)(str); // TODO - How do we know that this class expects a single str input?
```

Comment:
@mikehaverstock thoughts on this remaining todo, should we open a ticket to track?